src/buffer.cc

```diff
@@ -221,6 +221,21 @@ Result Buffer::CopyTo(Buffer* buffer) const {
   return {};
 }
 
+Result Buffer::IsEqual(Buffer* buffer) const {
+  if (buffer->buffer_type_ != buffer_type_)
+    return Result{"Buffers have a different type"};
+  if (buffer->size_ != size_)
+    return Result{"Buffers have a different size"};
+  if (buffer->width_ != width_)
+    return Result{"Buffers have a different width"};
+  if (buffer->height_ != height_)
+    return Result{"Buffers have a different height"};
+  if (buffer->values_ != values_)
```

Comment:
We probably want to do this the hard way. We want to be able to spit out two things, the first index where the buffers differ and their values and the number of indexes in which the buffers differ so, something along the lines of:

```
Buffers have different values. 1234 values differed, first difference at 12 values 7 != 8
```

Something like that.